pkg/dns/aws/dns.go

```diff
@@ -46,6 +47,8 @@ const (
 var (
 	_   dns.Provider = &Provider{}
 	log              = logf.Logger.WithName("dns")
+
+	hostedZoneIDRegex = regexp.MustCompile("^/?hostedzone/(.+)$")
```

Comment:
Can the hosted zone id have a slash?  I'm asking because the old logic did not permit slashes in hosted zone ids, but this regexp does, and I'm just curious if the change was deliberate.

Reply:
Yes, the change is deliberate.
The ID in the output from `ListHostedZones` starts with a slash.
```
$ aws route53 list-hosted-zones --max-items=1 | jq '.HostedZones[].Id'
"/hostedzone/Z0799914V7ZYCRPWCWYK"
```
The resource ARN in the output from `GetResources` does not have a slash before "hostedzone".
```
$ aws resourcegroupstaggingapi get-resources --resource-type-filters route53:hostedzone --max-items=1 | jq ".ResourceTagMappingList[].ResourceARN"
"arn:aws:route53:::hostedzone/Z03898872UPH0MB9MUBXE"
```

Reply:
(I could be wrong) but I think @Miciah was referring to how this regex permits slashes _in_ the hosted zone ID (the part after `/hostedzone/`), ie `/hostedzone/abc/123`. From my understanding, the prior code splits the entire string on `/`'s, and only allows 2 results https://github.com/openshift/cluster-ingress-operator/pull/530/files#diff-1a81cbdf88bfc6111c38c87b8d2c9d360f6e44f4b76b1f5c11435a3fd3c47cfaL261.

Reply:
Ah, sorry. I misunderstood. I can add restrictions that will error on extra slashes.